tests/client/core/test_searchUtils.js

```diff
@@ -16,3 +21,77 @@ describe('searchUtils mapStateToProps()', () => {
     assert.deepEqual(props, { hasSearchParams: false });
   });
 });
+
+describe('searchUtils loadCategoryResultsIfNeeded()', () => {
+  const state = {
+    api: { clientApp: 'firefox', lang: 'fr-CA' },
+    addons: {
+      ab: { slug: 'ab', name: 'ad-block' },
+      cd: { slug: 'cd', name: 'cd-block' },
+    },
+    search: {
+      filters: {
+        addonType: 'extension',
+        category: 'ad-block',
+        clientApp: 'firefox',
+      },
+      hasSearchParams: true,
+      loading: false,
+      results: [{ slug: 'ab', name: 'ad-block' }],
+    },
+  };
+
+  it('returns right away when loaded', () => {
+    const page = 10;
+    const stateWithPage = deepcopy(state);
+    stateWithPage.search.page = page;
+    const store = {
+      dispatch: sinon.spy(),
+      getState: () => stateWithPage,
+    };
+    const location = { query: { page } };
+    const params = {
+      addonType: 'extension',
+      application: 'firefox',
+      slug: 'ad-block',
+    };
+    assert.strictEqual(
+      loadCategoryResultsIfNeeded({ store, location, params }), true);
```

Comment:
I think this test would be better if it verified that `mockApi` is never called, which is a stronger indicator that data is not fetched from the API.